src/autodiff.jl

```diff
@@ -319,68 +382,135 @@ function uncompress_kernel!(J::CUSPARSE.CuSparseMatrixCSR, compressedJ, coloring
-            view(arrays.t1sx, nvbus+1:2*nvbus),
+            J.t1sF,
+            view(J.t1sx, 1:nvbus),
+            view(J.t1sx, nvbus+1:2*nvbus),
             ybus_re, ybus_im,
             pinj, qinj,
             pv, pq, nbus
         )
     elseif isa(type, ControlJacobian)
         residual_polar!(
-            arrays.t1sF,
-            view(arrays.t1sx, 1:nvbus),
+            J.t1sF,
+            view(J.t1sx, 1:nvbus),
             v_a,
             ybus_re, ybus_im,
-            view(arrays.t1sx, nvbus+1:nvbus+ninj), qinj,
+            view(J.t1sx, nvbus+1:nvbus+ninj), qinj,
             pv, pq, nbus
         )
     else
         error("Unsupported Jacobian structure")
     end
 
-    getpartials_kernel!(arrays.compressedJ, arrays.t1sF, nbus)
-    uncompress_kernel!(arrays.J, arrays.compressedJ, arrays.coloring)
+    getpartials_kernel!(J.compressedJ, J.t1sF, nbus)
+    uncompress_kernel!(J.J, J.compressedJ, J.coloring)
 
     return nothing
 end
 
+"""
+    residual_hessian_adj_tgt!(H::Hessian,
+                        residual_adj_polar!,
+                        lambda, tgt,
+                        vm, va, ybus_re, ybus_im, pinj, qinj, pv, pq, ref,
+                        nbus)
+
+Update the sparse Jacobian entries using AutoDiff. No allocations are taking place in this function.
+
+* `H::Hessian`: Factory created Jacobian object to update
+* `residual_adj_polar`: Adjoint function of residual
+* `lambda`: Input adjoint, usually lambda from a Langrangian
+* `tgt`: A tangent direction or vector that the Hessian should be multiplied with
+* `vm, va, ybus_re, ybus_im, pinj, qinj, pv, pq, ref, nbus`: Inputs both
+  active and passive parameters. Active inputs are mapped to `x` via the preallocated views.
+"""
+function residual_hessian_adj_tgt!(H::Hessian,
+                             residual_adj_polar!,
+                             lambda, tgt,
+                             v_m, v_a, ybus_re, ybus_im, pinj, qinj, pv, pq, ref, nbus)
+    x = H.x
+    ntgt = length(tgt)
+    nvbus = length(v_m)
+    ninj = length(pinj)
+    t1sx = H.t1sx
+    adj_t1sx = similar(t1sx)
+    t1sF = H.t1sF
+    adj_t1sF = similar(t1sF)
+    x[1:nvbus] .= v_m
+    x[nvbus+1:2*nvbus] .= v_a
+    x[2*nvbus+1:2*nvbus+ninj] .= pinj
+    t1sx .= H.x
+    adj_t1sx .= 0.0
+    t1sF .= 0.0
+    adj_t1sF .= lambda
+    # Seeding
+    nmap = length(H.map)
+    t1sseedvec = zeros(Float64, length(x))
+    for i in 1:nmap
+        t1sseedvec[H.map[i]] = tgt[i]
+        H.t1sseeds[i] = ForwardDiff.Partials{1, Float64}(NTuple{1, Float64}(tgt[i]))
+        t1sseedvec .= 0.0
+    end
+    seed_kernel!(H.t1sseeds, H.varx, H.t1svarx, nbus)
+    residual_adj_polar!(
+        t1sF, adj_t1sF,
+        view(t1sx, 1:nvbus), view(adj_t1sx, 1:nvbus),
+        view(t1sx, nvbus+1:2*nvbus), view(adj_t1sx, nvbus+1:2*nvbus),
+        ybus_re, ybus_im,
+        view(t1sx, 2*nvbus+1:2*nvbus+ninj), view(adj_t1sx, 2*nvbus+1:2*nvbus+ninj),
+        qinj, zeros(eltype(qinj), length(qinj)),
```

Comment:
here, we are allocating an array with size `length(qinj)`. Is this expected?

Reply:
I think so. These are the adjoint of `qinj` which there are also `length(qinj)`.

Reply:
I revise this. You are right, we never compute the adjoint of `qinj`, so removed it.